### ISSUE TITLE
Create new document loading bar to show that something is being done after a user clicks on create document

### ISSUE THREAD

**Issue by d2fong** [PathwayCommons/factoid#171]

**Comment by d2fong** [comment]
@sacdallago 

There is already a loading indicator right next to the ```create document```  but maybe it is not as noticeable as a progress bar.

![screen shot 2018-01-31 at 9 56 35 am](https://user-images.githubusercontent.com/2328291/35631082-f830682c-0670-11e8-893d-8ce1118e6c6f.png)


I can try to add a progress bar, but it will probably be an pseudo measure of progress.  I think from the client's point of view, I only know that the ```new doc``` request is in progress or if it is finished.

**Comment by sacdallago** [comment]
Ok. So I did some testing:

1. That loading bar doesn't show on button click but after **some time**. As you pointed out, and from what I understand, it gets triggered when a response from the server is received and a new document is being created. This is not what should happen. You can either keep it this way, and add a chip that pops up when the button is clicked that says something in the direction of "you request is being elaborated, you'll see a document shortly" or activate the spinning circle also between the time the user clicks the button and a response is obtained.


2. If I use some random abstract text, there is absolutely no feedback to the user. After a considerable amount of time (way more than 30 seconds, in which the user has no idea what's going on), I get a `console.error` (which the average user will never see). I attach stack trace, text used and gif (that stops after 30secs).

Stack trace:
```javascript
deps.js:30 Unhandled rejection SyntaxError: Unexpected token < in JSON at position 0
SyntaxError: Unexpected token < in JSON at position 0
    at i (http://factoid.baderlab.org/bundle.js:1:123596)
    at o (http://factoid.baderlab.org/deps.js:30:72778)
    at o._settlePromiseFromHandler (http://factoid.baderlab.org/deps.js:30:46184)
From previous event:
    at o.d [as _captureStackTrace] (http://factoid.baderlab.org/deps.js:30:10250)
    at o._then (http://factoid.baderlab.org/deps.js:30:41631)
    at o.then (http://factoid.baderlab.org/deps.js:30:39987)
    at t.value (http://factoid.baderlab.org/bundle.js:1:123831)
    at onClick (http://factoid.baderlab.org/bundle.js:1:124083)
    at Object.r (http://factoid.baderlab.org/deps.js:30:193170)
    at a (http://factoid.baderlab.org/deps.js:30:191519)
    at Object.s [as executeDispatchesInOrder] (http://factoid.baderlab.org/deps.js:30:191734)
    at h (http://factoid.baderlab.org/deps.js:30:125967)
    at v (http://factoid.baderlab.org/deps.js:30:126093)
    at Array.forEach (<anonymous>)
    at r (http://factoid.baderlab.org/deps.js:53:225499)
    at Object.processEventQueue (http://factoid.baderlab.org/deps.js:30:127155)
    at r (http://factoid.baderlab.org/deps.js:53:513241)
    at Object.handleTopLevel [as _handleTopLevel] (http://factoid.baderlab.org/deps.js:53:513311)
    at o (http://factoid.baderlab.org/deps.js:53:535414)
    at r.perform (http://factoid.baderlab.org/deps.js:30:148817)
    at Object.batchedUpdates (http://factoid.baderlab.org/deps.js:53:534963)
    at Object.o [as batchedUpdates] (http://factoid.baderlab.org/deps.js:30:83347)
    at dispatchEvent (http://factoid.baderlab.org/deps.js:53:536269)
```

Example text (notice that sometimes numbers pop up: these are citations in the abstract which can be considered as errors, but tbh I don't expect people to do the effort of removing them. So, I would say this is a pretty average-user like situation):
> Induced pluripotent stem (iPS) cells can be generated from somatic cells by introduction of Oct3/4, Sox2, Klf4 and c-Myc, in mouse1-4 and human5-8. Efficiency of this process, however, is low9. Pluripotency can be induced without c-Myc, but with even lower efficiency10,11. A p53 siRNA was recently shown to promote human iPS cell generation12, but specificity and mechanisms remain to be determined. Here we report that up to 10% of transduced mouse embryonic fibroblasts (MEF) lacking p53 became iPS cells, even without the Myc retrovirus. The p53 deletion also promoted induction of integration-free mouse iPS cells with plasmid transfection. Furthermore, in the p53-null background, iPS cells were generated from terminally differentiated T lymphocytes. Suppression of p53 also increased the efficiency of human iPS cell generation. DNA microarray analyses identified 34 p53-regulated genes that are common in mouse and human fibroblasts. Functional analyses of these genes demonstrate that the p53-p21 pathway serves as a safeguard not only in tumorigenicity, but also in iPS cell generation.


![feb-01-2018 09-35-59](https://user-images.githubusercontent.com/6240943/35684044-b646cbe0-0733-11e8-897e-5bb31e246c5b.gif)


In conclusion, there are few improvements I'd suggest:

1. Spinning wheel as soon as the user clicks the button until a document is produced
2. Catch eventual errors and display an appropriate message to the user
3. EVENTUALLY: decide upon a TTL for the request sent out to REACH (the question here is really: how long can we expect a request to be?)

**Comment by d2fong** [comment]
Okay thanks for the research.  

So from what I gather maybe:
* a empty card with a loading circle inside it with the message ```generating document```
* if the request fails, display an error message/tell the user that something went wrong ( i think it makes sense to display this inside the card)
* if the request succeeds, display the clickable document links like we currently do now

In terms of TTL, I think it depends on how large the input is.  I believe most of the previous testing on Factoid has been mostly with abstracts.  The time to process a whole paper will probably be longer.

**Comment by sacdallago** [comment]
I like this approach! What do you exactly mean by:
> i think it makes sense to display this inside the card

And:
> In terms of TTL, I think it depends on how large the input is. I believe most of the previous testing on Factoid has been mostly with abstracts. The time to process a whole paper will probably be longer.

We could display different messages for different lengths of text with a ballpark estimate. At this stage, I'm also tempted to throw in the idea of giving users the possibility to set an email address and get the links per email once reach has completed it's run. But, for now, let's focus on the doable.

I think for REACHes response time estimate maybe @fdurupinar has an idea?

**Comment by fdurupinar** [comment]
REACH response time is somehow  inconsistent. I uploaded a paper of 54 pages and had to cancel it after waiting for a response for 25 min.  On the other hand, a paper of 31 pages just took a few seconds to respond.

**Comment by sacdallago** [comment]
Thanks @fdurupinar . I'm getting a feeling that ultimately REACH is not the best choice for us, given that it presents very bad bugs, like not being able to predict if it will work or not, how long it will take, and ambiguity in the responses: https://github.com/PathwayCommons/factoid/issues/179#issuecomment-363574561

**Comment by sacdallago** [comment]
This seems to be addressed.
